extensions/docker-desktop/Makefile

```diff
@@ -16,64 +19,116 @@ BUILDER=buildx-multi-arch
+	make -j build-downloader build-tanzu-cli build-apps-bin build-client
 
-build-extension: ## Build the extension container image
-	docker build $(BUILD_ARGS) --tag=$(EXTENSION_IMAGE):$(TAG) --build-arg TAG=$(TAG) --build-arg REGISTRY=$(REGISTRY) --build-arg ORG=${ORG} --build-arg IMAGE_NAME=${IMAGE_NAME} .
 .PHONY: build-extension
+build-extension: build-deps ## Build the extension image for local architecture
+	docker buildx build $(BUILD_ARGS) --tag=$(EXTENSION_IMAGE):$(TAG) --build-arg TAG=$(TAG) --build-arg REGISTRY=$(REGISTRY) --build-arg ORG=${ORG} --build-arg IMAGE_NAME=${IMAGE_NAME} .
+##### BUILD
 
-extension: ## Build the extension and its dependencies
-	make build-deps build-extension
-.PHONY: extension
-
+##### DOCKER
+.PHONY: install
 install: ## Install the extension or update it if already exists
 	docker extension install $(EXTENSION_IMAGE):$(TAG) || docker extension update $(EXTENSION_IMAGE):$(TAG)
 
-update: ## Update the extension with a new image
+.PHONY: build-install 
+build-install: build-extension install ## Build extension images for local arch and install
+
+.PHONY: update
+update: ## Pull and update the extension with a new image
 	docker pull $(EXTENSION_IMAGE):$(TAG) && docker extension update $(EXTENSION_IMAGE):$(TAG)
 
+.PHONY: debug
 debug: ## Enable debug in the extension
 	docker extension dev debug $(EXTENSION_IMAGE):$(TAG)
 
-source: ## Replace the ui source of the extension
+.PHONY: source
+source: ## Replace the UI source of the extension
 	docker extension dev ui-source $(EXTENSION_IMAGE):$(TAG) $(DEV_UI_SOURCE)
 
+.PHONY: validate
 validate: ## Validate the extension
 	docker extension validate $(EXTENSION_IMAGE):$(TAG)
 
+.PHONY: dev-reset
 dev-reset: ## Reset development status of the extension
 	docker extension dev reset $(EXTENSION_IMAGE):$(TAG)
 
+.PHONY: delete
 delete: ## Remove the extension
 	docker extension rm $(EXTENSION_IMAGE):$(TAG)
+##### DOCKER
 
-prepare-buildx: ## Create buildx builder for multi-arch build, if not exists
+##### PUBLISH
+.PHONY: prepare-buildx
+prepare-buildx: 
 	docker buildx inspect $(BUILDER) || docker buildx create --name=$(BUILDER) --driver=docker-container --driver-opt=network=host
 
-push-extension: prepare-buildx ## Build & upload extension image to a registry
-	# Do not push if tag already exists: make push-extension tag=0.1
-	docker pull $(EXTENSION_IMAGE):$(tag) && echo "Failure: Tag already exists" || docker buildx build --push --builder=$(BUILDER) --platform=linux/amd64,linux/arm64 --build-arg TAG=${tag)} --tag=$(EXTENSION_IMAGE):$(tag) .
+stop-buildx:
+	docker buildx stop $(BUILDER)
+
+.PHONY: push-extension
+push-extension: prepare-buildx
+	# Do not push if tag already exists: TAG=0.1.0 make push-extension
+	docker pull $(EXTENSION_IMAGE):$(TAG) && echo "Failure: Tag already exists" || docker buildx build --push --builder=$(BUILDER) --platform=$(PLATFORMS) --build-arg TAG=$(TAG) --tag=$(EXTENSION_IMAGE):$(TAG) .
```

Comment:
I wonder if we need some sort of "safety" on here to prevent someone new coming in and poking around accidentally running this target and publishing something when they didn't really mean to. I think in our release automation we look for a value configured in the GitHub Action. So maybe an environment variable or make flag that we can use to protect this?

Reply:
True true.

In theory, only people who are in the VMware org in Dockerhub would be able to do this anyways, but yes, we should put up a check for even those people 👍🏼

Reply:
Thanks for adding this in.

Today I was saddened to learn that [DockerHub can't provide tag-protection](https://github.com/docker/roadmap/issues/85).

Reply:
Added a check where if a user is running the `build-push` targets, they must have the `TCE_CI_BUILD=true` env var (same as what our release automation does)